src/api/app/models/flag.rb

```diff
@@ -67,26 +93,13 @@ def to_s
     ret
   end
 
-  validates :flag, :presence => true
-  validates :position, :presence => true
-  validates_numericality_of :position, :only_integer => true
-
-  before_validation(:on => :create) do
-    if self.project
-      self.position = (self.project.flags.maximum(:position) || 0 ) + 1
-    elsif self.package
-      self.position = (self.package.flags.maximum(:position) || 0 ) + 1
-    end
+  def name
+    name = flag
+    name += "-#{repo}" unless repo.blank?
```

Comment:
name += "a" is actually
name = name.concat("a") - so it creates an iterim object which is then assigned

while name << "a" is just name.concat("a") - no extra assignment